packages/edit-site/src/components/header/feature-toggle/index.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { flow } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { MenuItem, withSpokenMessages } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { check } from '@wordpress/icons';
+
+function FeatureToggle( {
+	onToggle,
+	isActive,
+	label,
+	info,
+	messageActivated,
+	messageDeactivated,
+	speak,
+} ) {
+	const speakMessage = () => {
+		if ( isActive ) {
+			speak( messageDeactivated || __( 'Feature deactivated' ) );
+		} else {
+			speak( messageActivated || __( 'Feature activated' ) );
+		}
+	};
+
+	return (
+		<MenuItem
+			icon={ isActive && check }
+			isSelected={ isActive }
+			onClick={ flow( onToggle, speakMessage ) }
+			role="menuitemcheckbox"
+			info={ info }
+		>
+			{ label }
+		</MenuItem>
+	);
+}
+
+export default compose( [
+	withSelect( ( select, { feature } ) => ( {
```

Comment:
Nit: but we're preferring useSelect/useDispatch lately

Reply:
Thanks! Refactored in 093b683d79e261c0e53823ef0b6731b2bb9f998c.